checkstyle-tester/projects-to-test-on.properties

```diff
@@ -14,7 +14,7 @@ guava|git|https://github.com/google/guava|v28.2||
 
 #spotbugs|git|https://github.com/spotbugs/spotbugs|3.1.2||
 #pmd|git|https://github.com/pmd/pmd|pmd_releases/6.21.0|**/pmd/pmd-java/src/test/**/*,**/pmd/cpd/files/*
-#spoon|git|https://github.com/INRIA/spoon.git|spoon-core-10.1.1|**/src/test/resources/**/*
+#spoon|git|https://github.com/INRIA/spoon.git|spoon-core-10.1.0|**/src/test/resources/**/*
```

Comment:
It looks like you are using the launch_diff.sh script, problem lies in the script not fetching all tags:
```
➜  temp git clone https://github.com/INRIA/spoon.git
Cloning into 'spoon'...
remote: Enumerating objects: 62816, done.
remote: Counting objects: 100% (13/13), done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 62816 (delta 2), reused 3 (delta 1), pack-reused 62803
Receiving objects: 100% (62816/62816), 32.23 MiB | 12.63 MiB/s, done.
Resolving deltas: 100% (39701/39701), done.
➜  temp cd spoon 
➜  spoon git:(master) git fetch --all --tags          
➜  spoon git:(master) git tag | grep spoon-core-10.1.1
spoon-core-10.1.1

```

Reply:
Also, CI is green in the main repo, so diff.groovy handles it: https://app.circleci.com/pipelines/github/checkstyle/checkstyle/16845/workflows/5b255c99-1659-4be6-8e5c-22b251ded8b4/jobs/226429?invite=true#step-103-218

```
Resetting git sources to commit 'spoon-core-10.1.1'
Running command: git reset --hard spoon-core-10.1.1
HEAD is now at 0821a745 10.1.1

```

Reply:
Yes to the script I am using. It is core of my regression setup.

> lies in the script not fetching all tags

I confirmed the same behavior on windows, haven't confirmed running `git fetch --all --tags` on my regression rig yet.

I am slightly confused. Doesn't a fetch by default pull in tags as well? I have the 10.2 tag and the 10.1 which seems to me that I do get tags. I still think it isn't being pulled in because of `This commit does not belong to any branch on this repository`.

Also, even if I wasn't using the script, I have been here long enough that https://github.com/checkstyle/contribution/issues/180 would also effect me as I never delete the repos (especially since my version fetches updates).

Reply:
fetch does not get all tags by default, in my experience

Reply:
https://git-scm.com/docs/git-fetch
> By default, any tag that points into the histories being fetched is also fetched

This must be why as the tag doesn't point to anything in the history. A clone must force an all tags like you suggested.

> CI is green in the main repo, so diff.groovy handles it

Technically you can't compare the 2. CI always does a clone. My script also does a clone if the repo isn't already cloned. If the script was in CI it will be green too.

=====

So should I close this PR?

Reply:
I chose older version so it can be merged without doing extra regression, but I can do that if that is what we want to do.

Reply:
Let's close this PR, perhaps we need to create some issues to update/improve launch_diff.sh.

Reply:
I thought we agreed to deprecate launch_diff.sh as we do more in groovy

Reply:
It is deprecated from Checkstyle's perspective, however its issues are also reproducable in groovy's version.

Reply:
So we need to do in groovy `git fetch --all --tags`

Reply:
That is one way but if we are not doing https://github.com/checkstyle/contribution/issues/180 which I asked for a resolution on, then I think we should just move to a tag that still "exists" in the repo and ignore the fetch for now.

Reply:
Let's not get sidetracked in this PR; testing on a commit that doesn't live in main branch IS weird. Let's merge this and discuss #180 elsewhere. @rnveach do we have an issue created to deprecate launch script?